diff --git a/src/Data/Scientific/Internal.hs b/src/Data/Scientific/Internal.hs
index 8133de4..e6274c6 100644
--- a/src/Data/Scientific/Internal.hs
+++ b/src/Data/Scientific/Internal.hs
@@ -558,9 +558,7 @@ instance RealFrac Scientific where
     -- | @'floor' s@ returns the greatest integer not greater than @s@
     floor = whenFloating $ \c e ->
               if dangerouslySmall c e
-              then if c < 0
-                   then -1
-                   else 0
+              then 0
               else fromInteger (c `divInteger` magnitude (-e))
     {-# INLINABLE floor #-}
 
