diff --git a/src/Data/Scientific/Internal.hs b/src/Data/Scientific/Internal.hs
index 8133de4..f74e86b 100644
--- a/src/Data/Scientific/Internal.hs
+++ b/src/Data/Scientific/Internal.hs
@@ -890,7 +890,7 @@ scientificP = do
     S2 n z1 <- foldDigits stepC (S2 0 0)
     let s = S3 n z1 0
     S3 coeff z expnt <- (ReadP.satisfy (== '.') >> foldDigits stepF s)
-                        ReadP.<++ return s
+                        `mplus` return s
     let signedCoeff | pos       =   coeff
                     | otherwise = (-coeff)
     (ReadP.satisfy isE >>
